addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1501,6 +1505,11 @@ export class OdooEditor extends EventTarget {
         const contents = range.extractContents();
 
         setSelection(start, nodeSize(start));
+        const startLi = closestElement(start, 'li');
+        if (startLi && startLi.classList.contains('o_checked') &&
+            startLi.textContent === '' && closestElement(end, 'li') !== startLi) {
```

Comment:
What if we have `<li><a><br></a></li>` for instance? Then the closest element could be the `<a>`. Should we remove the class in that case? Either way, a test would be welcome for that scenario.

Reply:
As that is an inline tag and would be removed completely, it should remove the class. I have added the test case.

Reply:
How about `<li><h1><br></h1></li>` then? Thanks for having added the test case :+1:

Reply:
According to me it should remove the class. But let me know if you think otherwise.